src/ranged.cpp

```diff
@@ -118,17 +201,17 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
 
     tripoint target = target_arg;
     std::vector<tripoint> trajectory;
-    if( missed_by >= 1.0 ) {
+    if( missed_by_tiles >= 1.0 ) {
         // We missed enough to target a different tile
         double dx = target_arg.x - source.x;
         double dy = target_arg.y - source.y;
         double rad = atan2( dy, dx );
-        // Cap spread at 30 degrees or it gets wild quickly
-        double spread = std::min( ARCMIN( dispersion ), DEGREES( 30 ) );
-        rad += rng_float( -spread, spread );
+
+        // cap wild misses at +/- 30 degrees
+        rad += std::max( std::min( ARCMIN( shot_dispersion ), DEGREES( 30 ) ), DEGREES( -30 ) );
```

Comment:
That sounds good. It's not essential here (hence tangent) but would be a welcome addition in a separate PR if someone was to chef up a C++17 compliant implementation.